index.html

```diff
@@ -469,13 +493,220 @@ <h4>Attributes</h4>
+              frames duration</dfn> is the total duration of all [= dropped
+              audio frames =]. This measurement is incremented at the same time
+              as [= dropped audio frames =] and is measured in milliseconds.</p>
+            </li>
+            <div class="note">
+              <p>Because audio capture devices produce audio in real-time,
+              audio frames may be dropped if not processed in a timely
+              manner.</p>
+              <p>The percentage of audio that was dropped can be calculated as
+              [= dropped audio frames duration =] / ([= delivered audio frames
+              duration =] + [= dropped audio frames duration =]).</p>
+            </div>
+            <div class="note">
+              <p>If the track is unmuted and enabled, the counters increase as
+              audio is produced by the capture device. If no audio is flowing,
+              such as if the track is muted or disabled, then the counters do
+              not increase.</p>
+            </div>
+            </li>
+          </ul>
+          <p>Let the {{MediaStreamTrackAudioStats}} have internal slots
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\DeliveredFrames]]</dfn>,
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\DeliveredFramesDuration]]</dfn>,
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\DeliveredFramesLatency]]</dfn>,
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\Latency]]</dfn>,
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\DroppedFrames]]</dfn>,
+          and
+          <dfn class=export data-dfn-for="MediaStreamTrackAudioStats">[[\DroppedFramesDuration]]</dfn>,
+          initialized to 0.</p>
+          <p>The <dfn data-lt="expose audio frame counters steps">expose audio
+          frame counters steps</dfn> are the following:</p>
+          <ul>
+            <li>
+              <p>If this is the first time the [= expose audio frame counters
+              steps =] are called for this {{MediaStreamTrackAudioStats}} in
+              this task execution cycle, set the internal slots of as follows:
+              {{MediaStreamTrackAudioStats/[[DeliveredFrames]]}} is set to
+              [= delivered audio frames =],
+              {{MediaStreamTrackAudioStats/[[DeliveredFramesDuration]]}} is set
+              to [= delivered audio frames duration =],
+              {{MediaStreamTrackAudioStats/[[DeliveredFramesLatency]]}} is set to
+              [= delivered audio frames latency =],
+              {{MediaStreamTrackAudioStats/[[Latency]]}} is set to
+              [= current audio latency =],
+              {{MediaStreamTrackAudioStats/[[DroppedFrames]]}} is set to
+              [= dropped audio frames =] and
+              {{MediaStreamTrackAudioStats/[[DroppedFramesDuration]]}} is set to
+              [= dropped audio frames duration =].</p>
```

Comment:
Please rewrite this in terms of an algorithm running in parallel and stable state: https://html.spec.whatwg.org/multipage/webappapis.html#await-a-stable-state so it's clear what happen. Alternatively you can post a task from your algorithm running in parallel, that's more or less equivalent here.

No need to invent something else.

Reply:
Note https://github.com/whatwg/html/issues/2882.

Reply:
I already filed #108 but did not have a reference for the language requested. Now I have a reference, but I also have a bug that questions this wording (Jan-Ivar's link). So I'm not sure what to make of that, but let's follow up in that issue as a separate PR because we should fix both the audio and video algorithm at the same time, they use identical language and this is editorial.

Reply:
> I also have a bug that questions this wording (Jan-Ivar's link)

To clarify, Paul said:
1. "please rewrite this in terms of an algorithm running in parallel and stable state"
2. "Alternatively you can post a task from your algorithm running in parallel, that's more or less equivalent here.

My reason for mentioning https://github.com/whatwg/html/issues/2882 was to suggest going with 2.

They're just different ways to write algorithms in the spec that involve synchronous steps AND async steps.

To apply a JS analogy to spec writing: 1 is like async-await, and 2 is like promise-then. 1 is out of favor, because 2 is closer to implementations and c++.

Reply:
Ack, let's continue over at #108 where I quoted this last response.